src/data_conversion.rs

```diff
@@ -654,6 +656,71 @@ pub fn convert_battery_harvest(
         .collect()
 }
 
+#[cfg(feature = "zfs")]
+pub fn convert_arc_labels(current_data: &data_farmer::DataCollection) -> Option<(String, String)> {
+    /// Returns the unit type and denominator for given total amount of memory in kibibytes.
+    fn return_unit_and_denominator_for_mem_kib(mem_total_kib: u64) -> (&'static str, f64) {
+        if mem_total_kib < 1024 {
+            // Stay with KiB
+            ("KiB", 1.0)
+        } else if mem_total_kib < MEBI_LIMIT {
+            // Use MiB
+            ("MiB", KIBI_LIMIT_F64)
+        } else if mem_total_kib < GIBI_LIMIT {
+            // Use GiB
+            ("GiB", MEBI_LIMIT_F64)
+        } else {
+            // Use TiB
+            ("TiB", GIBI_LIMIT_F64)
+        }
+    }
+
+    if current_data.arc_harvest.mem_total_in_kib > 0 {
+        Some((
+            format!(
+                "{:3.0}%",
+                current_data.arc_harvest.use_percent.unwrap_or(0.0)
+            ),
+            {
+                let (unit, denominator) = return_unit_and_denominator_for_mem_kib(
+                    current_data.arc_harvest.mem_total_in_kib,
+                );
+
+                format!(
+                    "   {:.1}{}/{:.1}{}",
+                    current_data.arc_harvest.mem_used_in_kib as f64 / denominator,
+                    unit,
+                    (current_data.arc_harvest.mem_total_in_kib as f64 / denominator),
+                    unit
+                )
+            },
+        ))
+    } else {
+        None
+    }
+}
```

Comment:
nit: newline after function would be preferred here for a bit of visual clarity.